### PR TITLE
Fix UIATextInfo and MSHTMLTextInfo, updateCaret without selecting text

### LINKED ISSUE #13899: NVDA incorrectly selects text when performing a say all in Notepad
### Steps to reproduce:
1. Open Notepad and perform a say all, for example by typing text like this.
Test
Test

### Actual behavior:
NVDA selects text, and as a result, selected and unselected announcements occur repeatedly, and say all repeats lines.
### Expected behavior:
NVDA shouldn't select text in say all.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
alpha-25844,44086021
#### Windows version:
Windows 11, build 25151.1010
#### Name and version of other software in use when reproducing the issue:
Notepad 11.2205.11.0
#### Other information about your system:
N/A
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, this issue was first introduced in alpha-25838,82a4f861.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:
Fixes #13899 

### Summary of the issue:
Using Notepad on Windows 11, performing Say All (``nvda+downArrow|nvda+a``) selects each line of text.

This is because `UIATextInfo.updateCaret = UIATextInfo.updateSelection`, so updating the caret sets the selection.

The same problem exists with `MSHTMLTextInfo`

### Description of user facing changes
Say All no longer highlights each line of text.

### Description of development approach
1. Copy the text info
2. Collapse the new text info (e.g the range endpoints become (start, start))
3. Set the selection on the new range

### Testing strategy:
Test with STR in #13899 with Notepad
Test Say All with Internet Explorer as well.

### Change log entries:
Fixes unreleased regression

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

File: source/NVDAObjects/UIA/__init__.py [hunk 1]
```diff
@@ -923,7 +923,11 @@ def setEndPoint(self,other,which):
 	def updateSelection(self):
 		self._rangeObj.Select()
 
-	updateCaret = updateSelection
+	def updateCaret(self) -> None:
+		copyTextInfo = self.copy()
+		copyTextInfo.collapse()
```
Comment (feerrenrut) [inline source/NVDAObjects/UIA/__init__.py:928]: Why does self have a range? (I.E. not collapsed already?)
Is it to keep track of the old vs new position?
Does this behavior match other `TextInfo` implementations?

Comment (seanbudd) [reply source/NVDAObjects/UIA/__init__.py:928]: The behaviour I am introducing is the norm for other TextTnfos

https://github.com/nvaccess/nvda/blob/6f211a13fc1d9e2dc6400504173c314d917391a7/source/NVDAObjects/window/winword.py#L1052-L1056

https://github.com/nvaccess/nvda/blob/07add7e0862ff3340e49c426741f9010f1e341fe/source/NVDAObjects/window/edit.py#L790-L794

https://github.com/nvaccess/nvda/blob/b95743da959148924183effa98ffcc7f6e5f0349/source/cursorManager.py#L459-L463

After a non comprehensive search, I found one other instance which uses the current behaviour - MSHTML.
I have opted to fix this one as well to be consistent.
https://github.com/nvaccess/nvda/blob/17e52866e1dfc353edcc7dc9d466449e7920a155/source/NVDAObjects/IAccessible/MSHTML.py#L404-L408

**Comment by seanbudd** [comment]
Note that this issue can be reproduced with Internet Explorer as well

**Comment by seanbudd** [comment]
I've checked all instances of `updateCaret` being defined, I believe this is only an issue with UIATextInfo and MSHTML

File: source/NVDAObjects/UIA/__init__.py (hunk 1 above)
Comment (seanbudd) [reply source/NVDAObjects/UIA/__init__.py:928]: The range is intentionally a range selection as it being used in say all, which creates text infos in chunks to be spoken.

**Review by feerrenrut — APPROVED** [review]

**Pull request merged**